gis/osmium-tool/Portfile

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+PortGroup           boost 1.0
+
+github.setup        osmcode osmium-tool 1.13.1 v
+github.tarball_from archive
+revision            0
+
+categories          gis
+platforms           darwin
+maintainers         {@frankdean fdsd.co.uk:frank.dean} openmaintainer
+
+license             GPL-3+ MIT BSD Boost-1
+
+description         A command line tool for working with OpenStreetMap
+
+long_description    A multipurpose command line tool for working with \
+                    OpenStreetMap data based on the Osmium library
+
+checksums           rmd160  3b024c02f3a1ae1a31ab3ef7369579e3d13fe59e \
+                    sha256  d6273e2614d390d8444b767018b7023bdac3538cbe094d2799eee50b6f08cd03 \
+                    size    394224
+
+compiler.cxx_standard 2011
+
+depends_lib-append  port:expat \
+                    port:bzip2 \
+                    port:zlib
```

Comment:
according to upstream it also uses RapidJSON which they vendored. Ideally we want to use the version that is/can be installed by MacPorts so you'll likely want to add a dependency on `rapidjson` and check that it's used.

Reply:
`lz4` is an optional dependency but you should either add it here so that it will be always present or make sure that it is not be used, even if the user has it already installed (to make sure to have ReproducibleBuilds)

Reply:
I've included dependencies on the rapidjson and lz4 ports.

I've ensured the MacPorts rapidjson package is used for the build, by removing the included rapidjson source code after the patch phase.

I've also added a dependency on pandoc to build the man pages.

Reply:
I can exclude the embedded rapidjson source by patching the CMake configuration:

```
--- CMakeLists.txt~	2021-02-01 11:15:21.000000000 +0000
+++ CMakeLists.txt	2021-09-08 16:35:48.000000000 +0100
@@ -216,7 +216,7 @@
     ${PROJECT_BINARY_DIR}/osmium
 )
 
-include_directories(SYSTEM include)
+#include_directories(SYSTEM include)
 include_directories(${PROJECT_BINARY_DIR}/src)
 
 #-----------------------------------------------------------------------------

```

I've tested that the build fails in trace mode when the rapidjson port file is excluded from the build, being unable to find a rapidjson header,  and succeeds when it is included.

But deleting the rapidjson sources, which does feel very harsh, but which are really only temporary build files, is a much more certain way of ensuring it is not accidentally included in the build.

I'm very happy to go with your preference.  I don't feel strongly either way.